server/src/main/java/org/candlepin/resource/util/ConsumerBindUtil.java

```diff
@@ -86,6 +90,7 @@ public void handleActivationKeys(Consumer consumer, List<ActivationKey> keys)
             handleActivationKeyOverrides(consumer, key.getContentOverrides());
             handleActivationKeyRelease(consumer, key.getReleaseVer());
             keySuccess &= handleActivationKeyServiceLevel(consumer, key.getServiceLevel(), key.getOwner());
+            consumerCurator.update(consumer);
```

Comment:
Don't we just want to update the consumer once?  I.e. outside the for loop?

Reply:
We need this here because either bind will lockAndLoad the consumer and wipe out the settings that were transferred from the activation key.

Reply:
Got it.  Didn't realize the `handleActivationKey...` methods had side-effects.